.travis.yml

```diff
@@ -1,77 +1,19 @@
 env:
-  - BUILD_TYPE=Release
-  - BUILD_TYPE=Debug
+  - BUILD_TYPE=release
+  - BUILD_TYPE=debug
 
-os: linux
-
-
-sudo: false
+language: cpp
 
+services:
+  - docker
 
-cache:
-  apt: true
-  directories:
-     - programs
 notifications:
   email: false
 
-addons: &gcc5
-  apt:
-    sources:
-    - ubuntu-toolchain-r-test
-    packages:
-    - build-essential
-    - libblas-dev
-    - liblapack-dev
-    - libboost-dev
-    - doxygen
-#    - graphviz
-    - g++-5
-    - gcc-5
-    - gfortran
-
-compiler: gcc-5
-language: cpp
-
 before_install:
-  - export PRG=$PWD/programs
-  - export CASA=$PWD
-  - export CXX=g++-5
-  - export CC=gcc-5
-  - ./scripts/setup_astyle.sh
-  - export PATH=$PWD/programs/astyle/build/gcc/bin:$PATH
-  - ./scripts/check_indentation.sh
-  - export TRILINOS_DIR=$PWD/programs/trilinos
-  - export DEAL_II_DIR=$PWD/programs/dealii
-  - export DEAL2LKIT_DIR=$CASA/deal2lkit-$BUILD_TYPE
-  - export P4EST_DIR=$PWD/programs/p4est
-  - export PATH=$PWD/programs/astyle/build/gcc/bin:$PATH
-  - export PATH=$PRG/numdiff/bin:$PATH
-  - ./scripts/travis/install_mpi.sh
-  - export PATH=$PRG/mpich/bin:$PATH
-  - ./scripts/travis/install_cmake.sh
-  - export PATH=$PRG/cmake/bin:$PATH
-  - ./scripts/travis/install_ninja.sh
-  - export PATH=$PRG/ninja:$PATH
-  - ./scripts/travis/install_numdiff.sh
-  - export CXX=$PRG/mpich/bin/mpicxx
-  - export CC=$PRG/mpich/bin/mpicc
-  - ./scripts/travis/install_p4est.sh
-  - ./scripts/travis/install_sundials.sh
-  - export SUNDIALS_DIR=$PRG/sundials
+- docker pull heltai/deal2lkit:$BUILD_TYPE
+- docker run -P -v `pwd`:/home/dealii/pi-DoMUS:rw heltai/deal2lkit:$BUILD_TYPE /bin/sh -c "cd pi-DoMUS; ./scripts/check_indentation.sh"
 
-install: true
-  
 script:
-  - ./scripts/travis/install_trilinos_OR_dealii.sh
-  - ./scripts/travis/install_deal2lkit.sh
-  - mkdir build
-  - cd build
-  - cmake -GNinja -DCMAKE_BUILD_TYPE=$BUILD_TYPE -DENABLE_ALL_EXECUTABLES=ON ..
-  - ninja -j2
-  - ctest -j2 -V -R "${BUILD_TYPE,,}"
-
-
-branches:
-  only:
-    - master
+- docker run -P -v `pwd`:/home/dealii/pi-DoMUS:rw heltai/deal2lkit:$BUILD_TYPE /bin/sh -c "mkdir pi-DoMUS/build; cd pi-DoMUS/build; cmake -GNinja -DENABLE_DOCUMENTATION=OFF  -DENABLE_ALL_EXECUTABLES=ON ../; ninja"
```

Comment:
if you don't specify anything, ninja uses all available processors by default. Do we know in advance how many are assigned to our travis boxes?